### DISCUSSION TITLE
Should Nuxt 3 Server API implement the CSRF protection?

### DISCUSSION THREAD

**Discussion by nakorndev** [nuxt/nuxt#19047] [Questions]
I don't know much about CSRF protection. I know that only if I do Express.js with EJS should make a CSRF token into a hidden input form to prevent this attack.

I wonder about my website. If I did `/pages/sign-up.vue` and this call to `/server/api/sign-up.post.ts` the API should implement CSRF protection or not need to? Or this attack can be prevented by another method like some HTTP Header or some new web browser version prevented this attack by default?

As I say I don't know much. I just thinking about implementing my own session in H3 and make some random CSRF token.

**Comment by Baroshem** [comment]
Hey,

Nuxt-Auth package automatically implements the CSRF Token. Maybe you could check that for inspiration on how it could be done in your project? Or just use this package generally as it works really well :)

In general, using a CSRF Token is a recommended approach for modern applications. You can read more about it [here](https://owasp.org/www-community/attacks/csrf) and [here](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html) and [here](https://www.youtube.com/watch?v=eWEgUcHPle0)
